internal/controlplane/xds_routes.go

```diff
@@ -134,6 +135,14 @@ func buildPolicyRoutes(options *config.Options, domain string) []*envoy_config_r
 			})
 		}
 
+		requestHeadersToRemove := policy.RemoveRequestHeaders
+		if !policy.PassIdentityHeaders {
```

Comment:
Rather than stripping the JWT/Claim related headers here, we should avoid setting them in the authorize service's `Check()` call entirely.  See:

https://github.com/cuonglm/pomerium/blob/0d9b7d28faf61f018c03f8558bc199ed28f5eabe/authorize/grpc.go#L27
https://github.com/cuonglm/pomerium/blob/0d9b7d28faf61f018c03f8558bc199ed28f5eabe/authorize/grpc.go#L113
https://github.com/cuonglm/pomerium/blob/0d9b7d28faf61f018c03f8558bc199ed28f5eabe/authorize/check_response.go#L31

The header removal pattern in envoy is only needed for the cookie so we don't leak it to the upstream service.  We always have to return it, unlike the other headers.

Reply:
@travisgroth I thought about it, too, but can't find a way to know whether the policy is turn on/off pass identity headers in authorize's `Check()`.

Reply:
Hmm.  I see what you're saying.  All the per-route logic is inside OPA now.  I was thinking of this as a global option.

We could attach the headers to the input into OPA with a new field on the `Request` struct.  Thoughts?

Reply:
I think this is reasonable policy to leave outside of our authz policy (which can/will be specific to each user's authorization policy). I think removing headers here makes sense to me because it feels more related to proxying (and routes) than authorization policy per-se (though it does use the header returned from AuthZ).

... does that make sense? @cuonglm @travisgroth

Reply:
I do feel strange putting the data into the authz response just to remove it in lua later.  I'm not sure there's a great answer here, though.

Reply:
I'm quite ok with doing it here for now.

The Authz should always make a response with its desired headers, because one can interact directly with the Authz instead of via proxy. Technically, it's will be strange if authorization response has no related identity information in the response headers. In our current situation, the proxy is the only one which consume Authz response, so it can do whatever it wants. For now, we just remove it per policy basis, but we can do what ever we want here in the future, not only just removing.